metadata-io/src/main/java/com/linkedin/metadata/service/BusinessAttributeUpdateHookService.java

```diff
@@ -76,58 +82,88 @@ public void handleChangeEvent(
     Urn urn = entityChangeEvent.getEntityUrn();
     log.info("Business Attribute update hook invoked for urn :" + urn);
 
-    RelatedEntitiesResult entityAssociatedWithBusinessAttribute =
-        graphService.findRelatedEntities(
+    Consumer<RelatedEntitiesScrollResult> consumer =
+        relatedEntitiesScrollResult -> {
+          try {
+            Set<Urn> entityUrns =
+                relatedEntitiesScrollResult.getEntities().stream()
+                    .map(RelatedEntity::getUrn)
+                    .map(UrnUtils::getUrn)
+                    .collect(Collectors.toSet());
+
+            Map<Urn, EntityResponse> entityResponseMap =
+                systemEntityClient.batchGetV2(
```

Comment:
2000 batchSize of Urns might break the URL length for using Rest.li, this might also need to utilize the OpenAPIClient if the batchSize really needs to be this large.